app/controllers/reservations_controller.rb

```diff
@@ -282,9 +283,14 @@ def load_basic_resources
     @order = Order.find(params[:order_id])
     # It's important that the order_detail be the same object as the one in @order.order_details.first
     @order_detail = @order.order_details.find { |od| od.id.to_i == params[:order_detail_id].to_i }
-    @reservation = @order_detail.reservation
-    @instrument = @order_detail.product
-    @facility = @instrument.facility
+    if @order_detail.present?
+      @reservation = @order_detail.reservation
+      @instrument = @order_detail.product
+      @facility = @instrument.facility
+    else
+      flash[:error] = text("order_detail_removed")
+      return redirect_to facility_path(@order.facility)
+    end
     nil
```

Comment:
Is there a reason this need to return `nil`? If we get rid of that, do we need the explicit `return` above?